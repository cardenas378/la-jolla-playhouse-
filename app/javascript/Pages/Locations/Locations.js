import PropTypes from "prop-types";
import React from "react";
import { Link, Redirect } from "react-router-dom";

import { getLocations, createLocation } from "../../components/api";
import { Header, AddLocation, AddLocationLink } from "../../components/index";
import { Location } from "../../components/Location";

class Locations extends React.Component {
  render() {
    const { locations } = this.props;

    return (
      <>
        <Header id="header" name="Locations">
          <AddLocationLink />
        </Header>
        {locations.map(location => {
          return (
            <Location
              contact_email={location.contact_email}
              contact_first_name={location.contact_first_name}
              contact_last_name={location.contact_last_name}
              contact_phone_number={location.contact_phone_number}
              id="location"
              key={location.id}
              location_type={location.location_type}
              locationID={location.id}
              name={location.name}
            />
          );
        })}
      </>
    );
  }
}

Locations.propTypes = {
  changeSuccess: PropTypes.func,
  conctact_email: PropTypes.string,
  contact_first_name: PropTypes.string,
  contact_last_name: PropTypes.string,
  contact_phone_number: PropTypes.string,
  id: PropTypes.number,
  location_type: PropTypes.string,
  locations: PropTypes.array,
  name: PropTypes.string
};

export { Locations };
