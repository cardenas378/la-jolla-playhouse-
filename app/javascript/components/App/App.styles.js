import styled from "styled-components";

const StyledLayout = styled.div`
  display: grid;
  grid-template-columns: 225px 1fr;
  grid-template-areas: "sidebar main";

  main {
    grid-area: main;
    background: ${props => props.theme.whiteSmoke};
    min-height: 100vh;
  }
`;

export { StyledLayout };
