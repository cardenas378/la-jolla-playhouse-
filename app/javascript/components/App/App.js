import PropTypes from "prop-types";
import React from "react";
import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Link
} from "react-router-dom";

import styled, { ThemeProvider } from "styled-components/macro";

import { Home } from "../../Pages/Home";
import { Locations } from "../../Pages/Locations";
import { AddBooking } from "../AddBooking";
import { AddLocation } from "../AddLocation";
import { getLocations, createLocation } from "../api";
import { NavBar } from "../NavBar";
import GlobalStyle from "../styles/Global";
import Theme from "../styles/Theme";

import { StyledLayout } from "./App.styles";

import {
  Header,
  AddBookingHeader,
  FormSectionHeader,
  StyledSelect,
  FullInput,
  RadioButton,
  SubmitButton,
  SaveBookingButton,
  HalfInput,
  ThirdInput
} from "..";

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      locations: [],
      success: false
    };
  }

  componentDidMount() {
    getLocations().then(locations => {
      this.setState({ locations });
    });
  }

  handleNewLocation = location => {
    createLocation(location).then(location => {
      this.setState({ success: true });
    });
  };

  changeSuccess = value => {
    this.setState({ success: value });
  };

  componentDidMount() {
    getLocations().then(locations => {
      this.setState({ locations });
    });
  }

  render() {
    const { locations, success } = this.state;

    return (
      <ThemeProvider theme={Theme}>
        <GlobalStyle />
        <StyledLayout>
          <Router>
            <NavBar />

            <main>
              <Route
                exact
                path="/AddLocation"
                render={props => {
                  return (
                    <AddLocation
                      {...props}
                      handleNewLocation={this.handleNewLocation}
                      success={success}
                    />
                  );
                }}
              />

              <Route
                exact
                path="/Locations"
                render={props => {
                  return (
                    <Locations
                      {...props}
                      changeSuccess={this.changeSuccess}
                      locations={locations}
                      success={success}
                    />
                  );
                }}
              />

              <Route
                exact
                path="/"
                render={props => {
                  return <Home {...props} />;
                }}
              />
              <Route
                exact
                path="/AddBooking/:location_id"
                render={props => {
                  return (
                    <AddBooking
                      {...props}
                      changeSuccess={this.changeSuccess}
                      locations={this.state.locations}
                      success={this.state.success}
                    />
                  );
                }}
              />
            </main>
          </Router>
        </StyledLayout>
      </ThemeProvider>
    );
  }
}

App.propTypes = {};

export { App };
