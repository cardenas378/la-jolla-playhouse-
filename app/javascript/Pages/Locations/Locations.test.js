import Enzyme, { mount, shallow, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router";

import { Locations } from "Pages/Locations";

import App from "../../components/App";

Enzyme.configure({ adapter: new Adapter() });

describe("Locations", () => {
  const defaultProps = {
    id: 1,
    name: "LEARN",
    location_type: null,
    contact_first_name: null,
    contact_phone_number: null,
    contact_email: null,
    contact_relationship: null,
    address_city: null,
    address_state: null,
    address_zip: null,
    phone_number: null,
    reference: null,
    school_principal: null,
    principal_email: null,
    school_district: null,
    county_district: null,
    city_district: null,
    created_at: "2019-11-04T17:11:04.945Z",
    updated_at: "2019-11-04T17:11:04.945Z",
    contact_last_name: null,
    address_1: null,
    address_2: null
  };

  const render = props => shallow(<Locations {...defaultProps} {...props} />);

  test("app renders with state objects", () => {
    const wrapper = render({ locations: [{ ...defaultProps }] });

    expect(wrapper).toMatchSnapshot();
  });
});
