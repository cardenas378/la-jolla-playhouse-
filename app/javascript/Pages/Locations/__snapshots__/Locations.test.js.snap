// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations app renders with state objects 1`] = `
<Fragment>
  <Header
    id="header"
    name="Locations"
  >
    <AddLocationLink />
  </Header>
  <Location
    contact_email={null}
    contact_first_name={null}
    contact_last_name={null}
    contact_phone_number={null}
    id="location"
    key="1"
    locationID={1}
    location_type={null}
    name="LEARN"
  />
</Fragment>
`;
